target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wr1043n-v5.c

```diff
@@ -0,0 +1,194 @@
+/*
+ * TP-Link TL-WR1043N v5 board support based on
+ * Atheros AP152 reference board support
+ *
+ * Copyright (c) 2014 The Linux Foundation. All rights reserved.
+ * Copyright (c) 2012 Gabor Juhos <juhosg@openwrt.org>
+ * Copyright (c) 2017 Tim Thorpe	<tim@tfthorpe.net>
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#include <linux/platform_device.h>
+#include <linux/ath9k_platform.h>
+#include <linux/ar8216_platform.h>
+#include <linux/proc_fs.h>
+#include <linux/delay.h>
+
+#include <asm/mach-ath79/ar71xx_regs.h>
+
+#include "common.h"
+#include "dev-m25p80.h"
+#include "machtypes.h"
+#include "pci.h"
+#include "dev-eth.h"
+#include "dev-gpio-buttons.h"
+#include "dev-leds-gpio.h"
+#include "dev-spi.h"
+#include "dev-wmac.h"
+
+
+#define TL_WR1043N_V5_GPIO_LED_WLAN             19
+#define TL_WR1043N_V5_GPIO_LED_WPS              1
+#define TL_WR1043N_V5_GPIO_LED_STATUS           6
+
+#define TL_WR1043N_V5_GPIO_LED_WAN_INET         15
+#define TL_WR1043N_V5_GPIO_LED_WAN_LINK         16
+#define TL_WR1043N_V5_GPIO_LED_LAN1             9
+#define TL_WR1043N_V5_GPIO_LED_LAN2             14
+#define TL_WR1043N_V5_GPIO_LED_LAN3             21
+#define TL_WR1043N_V5_GPIO_LED_LAN4             20
+
+
+#define TL_WR1043N_V5_GPIO_BTN_RESET            2
+#define TL_WR1043N_V5_GPIO_BTN_WPS_WIFI         5
+#define TL_WR1043N_V5_KEYS_POLL_INTERVAL        20     /* msecs */
+#define TL_WR1043N_V5_KEYS_DEBOUNCE_INTERVAL    (3 * TL_WR1043N_V5_KEYS_POLL_INTERVAL)
+
+#define TL_WR1043N_V5_MAC0_OFFSET               0
+#define TL_WR1043N_V5_MAC1_OFFSET               6
+#define TL_WR1043N_V5_WMAC_CALDATA_OFFSET       0x1000
+
+#define TL_WR1043N_V5_GPIO_MDC			3
+#define TL_WR1043N_V5_GPIO_MDIO			4
+
+extern void gpio_set_value(unsigned gpio, int value);
+
+
+static struct gpio_led tl_wr1043n_v5_leds_gpio[] __initdata = {
+	{
+		.name		= "tp-link:green:wlan",
+		.gpio		= TL_WR1043N_V5_GPIO_LED_WLAN,
+		.active_low	= 1,
+	},
+	{
+		.name		= "tp-link:green:system",
+		.gpio		= TL_WR1043N_V5_GPIO_LED_STATUS,
+		.active_low	= 1,
+		.default_state = LEDS_GPIO_DEFSTATE_ON,
+	},
+	{
+		.name		= "tp-link:green:lan_1",
```

Comment:
why '_1'? It seems most of the other board use lan1, lan2, etc. Might be a good idea to stick to the same naming conventions, especially across similar boards.